src/usr/share/gpgmailer/gpgmailer-daemon.py

```diff
@@ -437,23 +439,29 @@ def create_spool_directories(program_uid, program_gid):
         raise exception
 
     spool_dir = os.path.join(SYSTEM_SPOOL_DIR, PROGRAM_NAME)
-    mounted_as_tmpfs = check_if_mounted_as_tmpfs(spool_dir)
 
-    # If directory is not mounted as tmpfs and there is something in the directory, fail to
-    #   start.
-    if os.listdir(spool_dir) != [] and not mounted_as_tmpfs:
-        raise InitializationException('Program spool directory is not empty and not mounted '
-                                      'as a ramdisk. Startup failed.')
+    if config['use_tmpfs_spool']:
+        mounted_as_tmpfs = check_if_mounted_as_tmpfs(spool_dir)
 
-    # If the program spool directory is empty and not already mounted as tmpfs, mount it as
-    #   tmpfs.
-    if not mounted_as_tmpfs:
-        logger.info('Attempting to mount the program spool directory as a ramdisk.')
-        subprocess.call(['mount', '-t', 'tmpfs', '-o', 'size=25%', 'none', spool_dir])
+        # If directory is not mounted as tmpfs and there is something in the directory, fail
+        #   to start.
+        # TODO: Do we still want to make sure the directory is empty? (I'm thinking probably
+        #   not, although, nothing should be in there unless something goes wrong or the flag
+        #   has just been set to use tmpfs and something was just queued.)
+        if os.listdir(spool_dir) != [] and not mounted_as_tmpfs:
+            raise InitializationException(
+                'Program spool directory configured to be a ramdisk, but the directory is '
+                'not empty and not already mounted as a ramdisk. Startup failed.')
 
-    if not check_if_mounted_as_tmpfs(spool_dir):
-        raise InitializationException(
-            'Program spool directory was not mounted as a ramdisk. Startup failed.')
+        # If the program spool directory is empty and not already mounted as tmpfs, mount it
+        #   as tmpfs.
+        if not mounted_as_tmpfs:
+            logger.info('Attempting to mount the program spool directory as a ramdisk.')
+            subprocess.call(['mount', '-t', 'tmpfs', '-o', 'size=25%', 'none', spool_dir])
```

Comment:
Ramdisk size should probably be configurable, but this can be addressed when we merge parkbench-common's ramdisk module down.

Reply:
You can do it now if you want. If not, add a TODO and create bug.

Reply:
I chose to create a TODO+issue for now. There's no need to clutter up this project with more copied code when we are just going to delete it anyway.